db/populators/proceeding_type_service_levels_populator.rb

```diff
@@ -11,35 +11,33 @@ def self.call
   end
 
   def call
-    seed_data.each { |seed_row| populate(seed_row) }
+    seed_data_keys.each { |seed_hash| populate(seed_hash) }
   end
 
 private
 
-  def populate(seed_row)
-    record = ProceedingTypeServiceLevel.where(service_level_id: service_level_id(seed_row),
-                                              proceeding_type_id: proceeding_type_id(seed_row)).first_or_initialize
-    record.update! attributes(seed_row)
+  def populate(seed_data)
+    seed_data["proceeding_type_ids"].each do |proceeding_id|
+      record = ProceedingTypeServiceLevel.where(service_level_id: service_level_id(seed_data),
+                                                proceeding_type_id: proceeding_type_id(proceeding_id)).first_or_initialize
+      record.update!(
+        proceeding_type_id: proceeding_type_id(proceeding_id),
+        service_level_id: service_level_id(seed_data),
+        proceeding_default: seed_data["default"],
+      )
+    end
```

Comment:
Is may be worth avoiding the repeated calls to find service_level and find proceeding type just by using local variables. But since they are not DB calls i guess it makes little difference.

A bit of a name change I found clarified things for me below too: `proceeding_id` => `proceeding_ccms_code` or just `ccms_code`

```suggestion
    seed_data["proceeding_type_ids"].each do |proceeding_ccms_code|
        _service_level_id  = service_level_id(seed_data)
        _proceeding_type_id = proceeding_type_id(proceeding_ccms_code)
      
      record = ProceedingTypeServiceLevel
                        .where(service_level_id: _service_level_id,
                                     proceeding_type_id: _proceeding_type_id).first_or_initialize
                                     
      record.update!(
        proceeding_type_id: _proceeding_type_id ,
        service_level_id: _service_level_id,
        proceeding_default: seed_data["default"],
      )
    end
```